src/uct/api/uct.h

```diff
@@ -1329,6 +1390,29 @@ ucs_status_t uct_md_config_read(const char *name, const char *env_prefix,
                                 uct_md_config_t **config_p);
 
 
+
+/**
+ * @ingroup UCT_MD
+ * @brief Check if remote sock address is reachable from the memory domain.
+ *
+ * This function checks if a remote sock address can be reached from a local
+ * memory domain. Reachability can be checked in local or remote mode.
+ *
+ * @param [in]  md         Memory domain to check reachability from.
+ * @param [in]  sockaddr   Socket address to check reachability to.
+ * @param [in]  mode       Mode for checking reachability, as defined in @ref
+ *                         uct_sockaddr_reachability_t.
+ *                         Indicates if reachability is tested on the server side -
+ *                         for binding to the given sockaddr, or on the
+ *                         client side - for connecting to the given remote
+ *                         peer's sockaddr.
+ *
+ * @return Nonzero if reachable, 0 if unreachable.
+ */
+int uct_md_is_sockaddr_reachable(uct_md_h md, const ucs_sock_addr_t *sockaddr,
+                                 uct_sockaddr_reachability_t mode);
+
+
```

Comment:
for client side, need it to check which md to use for connecting (if, for example, we have tcp and rdma_cm)
for server side, in case user has provided an address to bind to (not INADDR_ANY), need to check which md's should be listening.
eventually it will check network mask, routing table, and so on, but UCP should not be aware of such details